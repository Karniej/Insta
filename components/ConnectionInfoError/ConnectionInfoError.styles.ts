import { StyleSheet } from 'react-native'

export default StyleSheet.create({
    networkErrorContainer: {
        paddingHorizontal: 10,
        paddingVertical: 8,
        backgroundColor: 'indianred',
        flexDirection: 'row',
        alignItems: 'flex-start'
    },
    networkError: {
        textTransform: 'uppercase',
        fontSize: 12,
        color: 'white',
        fontWeight: 'bold',
        textAlign: 'left',
        paddingLeft: 10
    }
})
