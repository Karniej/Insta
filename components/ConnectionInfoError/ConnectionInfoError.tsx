import React, { useRef, useEffect, useState } from 'react'
import AnimatedIcon from '../AnimatedIcon'
import { Text } from '../Themed'
import styles from './ConnectionInfoError.styles'
import * as Animatable from 'react-native-animatable'
import NetInfo from '@react-native-community/netinfo'

const ConnectionInfoError = () => {
  const [networkError, setNetworkError] = useState('')
  const shouldShowNetInfo = networkError?.length > 0

  useEffect(() => {
    const unsubscribe = NetInfo.addEventListener((info) => {
      const isConnected = info.type === 'cellular' || info.type === 'wifi'
      if (!isConnected) {
        setNetworkError("You are currently offline, the data is shown from your phone's cache")
      } else {
        setNetworkError('')
      }
    })
    return unsubscribe()
  }, [])

  return (
    <Animatable.View
      animation={shouldShowNetInfo ? 'slideInDown' : 'slideOutUp'}
      style={styles.networkErrorContainer}
    >
      <AnimatedIcon color='white' size={25} name='ios-information-circle-outline' />
      <Text style={styles.networkError}>{networkError}</Text>
    </Animatable.View>
  )
}

export default ConnectionInfoError
