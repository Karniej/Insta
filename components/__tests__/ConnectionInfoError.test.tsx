import * as React from 'react'
import ConnectionInfoError from '../ConnectionInfoError/ConnectionInfoError'
import { render, fireEvent } from '@testing-library/react-native'

it(`renders correctly`, () => {
  const tree = render(<ConnectionInfoError errorMessage='error' />).toJSON()

  expect(tree).toMatchSnapshot()
})
