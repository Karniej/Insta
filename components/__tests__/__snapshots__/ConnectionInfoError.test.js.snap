// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  forwardedRef={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "indianred",
      "flexDirection": "row",
      "paddingHorizontal": 10,
      "paddingVertical": 8,
    }
  }
>
  <Text />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "color": "white",
          "fontSize": 12,
          "fontWeight": "bold",
          "paddingLeft": 10,
          "textAlign": "left",
          "textTransform": "uppercase",
        },
      ]
    }
  >
    error
  </Text>
</View>
`;
